core/src/main/java/org/sonarsource/sonarlint/core/util/ws/WsRequest.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.util.ws;
+
+import java.util.Map;
+
+/**
+ * @since 5.3
+ */
+public interface WsRequest {
+
+  Method getMethod();
+
+  String getPath();
+
+  String getMediaType();
+
+  /**
+   *
+   * In case of multi value parameters, returns the first value
+   *
+   * @deprecated since 6.1. Use {@link #getParameters()} instead
+   */
+  @Deprecated
+  Map<String, String> getParams();
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)